testsuite/tests/basic/patmatch_split_no_or.ml

```diff
@@ -52,12 +52,12 @@ type t += A | B of unit | C of bool * int;;
 0
 type t = ..
 (let
-  (A/26 = (makeblock 248 "A" (caml_fresh_oo_id 0))
-   B/27 = (makeblock 248 "B" (caml_fresh_oo_id 0))
-   C/28 = (makeblock 248 "C" (caml_fresh_oo_id 0)))
-  (seq (apply (field_mut 1 (global Toploop!)) "A/26" A/26)
-    (apply (field_mut 1 (global Toploop!)) "B/27" B/27)
-    (apply (field_mut 1 (global Toploop!)) "C/28" C/28)))
+  (A = (makeblock 248 "A" (caml_fresh_oo_id 0))
+   B = (makeblock 248 "B" (caml_fresh_oo_id 0))
+   C = (makeblock 248 "C" (caml_fresh_oo_id 0)))
+  (seq (apply (field_mut 1 (global Toploop!)) "A/26" A)
+    (apply (field_mut 1 (global Toploop!)) "B/27" B)
+    (apply (field_mut 1 (global Toploop!)) "C/28" C)))
```

Comment:
I think the `"A/26"` is inherently more stable and trying to canonicalise that would be a risky special case, right?

Reply:
It's a string literal in the code, not an identifier name. This comes from the calls to `toploop_setvalue` in translmod.ml, which in turn call `Ident.unique_toplevel_name`, which is not affected by canonicalization. I think that we don't want `-dcanonical-ids` to affect the generated code, and also canonicalizing these "toplevel names" may be incorrect if the choices need to be consistent across distinct compiler invocations on modules and their dependencies.